src/antares/service/api_services/link_api.py

```diff
@@ -160,20 +163,57 @@ def update_link_ui(self, link: Link, ui: LinkUi) -> LinkUi:
             link_ui = LinkUi.model_validate(json_response)
 
         except APIError as e:
-            raise LinkUiUpdateError(link.name, e.message) from e
+            raise LinkUiUpdateError(link.id, e.message) from e
 
         return link_ui
 
-    def read_links(self) -> list[Link]:
-        raise NotImplementedError
+    def get_parameters(self, area_from_id: str, area_to_id: str) -> pd.DataFrame:
+        try:
+            parameters_path = f"input/links/{area_from_id}/{area_to_id}_parameters"
+            matrix = get_matrix(self._base_url, self.study_id, self._wrapper, parameters_path)
+        except APIError as e:
+            raise LinkDownloadError(f"{area_from_id}/{area_to_id}", "parameters", e.message)
```

Comment:
You should raise from e instead of giving the e.message. see other parts of the code